src/domains/coupon/__tests__/calculateCouponDiscount.test.ts

```diff
@@ -0,0 +1,173 @@
+import { CartItemWithSelection } from "../../cart/types/response";
+import * as basicCalculations from "../calculations/basic";
+import { calculateCouponDiscount } from "../calculations/combined/calculateCouponDiscount";
+import { Coupon, CouponType } from "../types/response";
+
+jest.mock("../calculations/basic", () => ({
+  calculateFixedDiscount: jest.fn().mockReturnValue(5000),
+  calculatePercentageDiscount: jest.fn().mockReturnValue(2000),
+  calculateBuyXGetYDiscount: jest.fn().mockReturnValue(10000),
+  calculateShippingDiscount: jest.fn().mockReturnValue(3000),
+}));
+
+describe("calculateCouponDiscount 함수 테스트", () => {
+  const orderPrice = 50000;
+  const shippingFee = 3000;
```

Comment:
이전에 비슷한 리뷰를 남겼을거 같기도 한데 큰 숫자를 사용할 땐 numeric separator를 이용해주시면 좋을거 같아요

```suggestion
  const orderPrice = 50_000;
  const shippingFee = 3_000;
```

Reply:
숫자하니까 생각난 소소한 팁인데 시간을 표현할 때는 보기 쉽게 * 도 많이 사용하니 요것도 참고하시면 좋겠네요

```
const minutes = 60 * 1_000 
const fiveMinutes = 5 * minutes
```

Reply:
> const minutes = 60 * 1_000 
const fiveMinutes = 5 * minutes

이 구조가 확실히 가독성이 좋아서 유지보수하기 편하겠네요! 감사합니다 😄